libs/dyn/pattern_trie.go

```diff
@@ -45,44 +42,42 @@ type trieNode struct {
 }
 
 // NewPatternTrie creates a new empty pattern trie.
-func NewPatternTrie() *PatternTrie {
-	return &PatternTrie{
-		root: &trieNode{},
-	}
+func NewPatternTrie() *TrieNode {
```

Comment:
Is this necessary? since empty TrieNode is usable, I can just create empty object.